File: InPlan.Web/Modules/ServerTypes/Parametros/TecnicoProyectoTecnologiasRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface TecnicoProyectoTecnologiasRow {
    IdTecnicoProyectoTecnologia?: number;
    IdTecnicoProyecto?: number;
    IdTecnologia?: number;
    IdTecnicoProyectoObservaciones?: string;
    IdTecnologiaDescripcion?: string;
}

export abstract class TecnicoProyectoTecnologiasRow {
    static readonly idProperty = 'IdTecnicoProyectoTecnologia';
    static readonly localTextPrefix = 'Parametros.TecnicoProyectoTecnologias';
    static readonly lookupKey = 'Parametros.TecnicoProyectoTecnologias';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<TecnicoProyectoTecnologiasRow>('Parametros.TecnicoProyectoTecnologias') }
    static async getLookupAsync() { return getLookupAsync<TecnicoProyectoTecnologiasRow>('Parametros.TecnicoProyectoTecnologias') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<TecnicoProyectoTecnologiasRow>();
}

File: InPlan.Web/Modules/Parametros/TecnicoProyectoTecnologias/TecnicoProyectoTecnologiasRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("TecnicoProyectoTecnologias")]
[DisplayName("Tecnico Proyecto Tecnologias"), InstanceName("Tecnico Proyecto Tecnologias")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed partial class TecnicoProyectoTecnologiasRow : Row<TecnicoProyectoTecnologiasRow.RowFields>, IIdRow
{
    const string jIdTecnicoProyecto = nameof(jIdTecnicoProyecto);
    const string jIdTecnologia = nameof(jIdTecnologia);

    class RowTemplate
    {
        [DisplayName("Id Tecnico Proyecto Tecnologia"), Identity, IdProperty]
        public int? IdTecnicoProyectoTecnologia { get; set; }

        [LookupEditor(typeof(TecnicosRow))]
        [DisplayName("Técnico"), NotNull, ForeignKey("TecnicosProyectos", "IdTecnicoProyecto"), LeftJoin(jIdTecnicoProyecto), TextualField(nameof(IdTecnicoProyectoObservaciones))]
        public int? IdTecnicoProyecto { get; set; }

        [LookupEditor(typeof(TecnologiasRow))]
        [DisplayName("Tecnología"), NotNull, ForeignKey("Tecnologias", "IdTecnologia"), LeftJoin(jIdTecnologia), TextualField(nameof(IdTecnologiaDescripcion))]
        public int? IdTecnologia { get; set; }

        [DisplayName("Observaciones"), Expression($"{jIdTecnicoProyecto}.[Observaciones]")]
        public string IdTecnicoProyectoObservaciones { get; set; }

        [DisplayName("Descripción"), Expression($"{jIdTecnologia}.[Descripcion]")]
        public string IdTecnologiaDescripcion { get; set; }
    }
}